### ISSUE TITLE
Settings to tolerate optional arguments

### ISSUE THREAD

**Issue by barraponto** [PyCQA/pylint#4903]
### Current problem

Consider the following case, where Dog.speak adds optional keyword arguments to the AnimalBase:

```python
class AnimalBase(ABC):

    @abstractmethod
    def say(self) -> str:
        ...


class Dog(AnimalBase):

    def say(self, loud=False) -> str:
        return "WOOF" if loud else "woof"
```

Right now, it emits `W0221: Number of parameters was 1 in 'AnimalBase.say' and is now 2 in overridden 'Dog.say' method (arguments-differ)`

### Desired solution

There should be a setting/flag to allow for extra optional arguments.

**Comment by Pierre-Sassoulas** [comment]
This code break the [Liskov substitution principle](https://en.wikipedia.org/wiki/Liskov_substitution_principle) which is the point of the check in the first place. You can disable ``arguments-differ`` if you want, but permitting optional argument would defeat the purpose of the check.

**Comment by Pierre-Sassoulas** [comment]
In my opinion LSP is stricter than that. From Wikipedia : ``Requirement: Let ϕ ( x ) be a property provable about objects x of type T. Then ϕ ( y ) should be true for objects y of type S where S is a subtype of T. ``. For me it means you should be able to use any subclass of ``AnimalBase`` interchangeably which you won't be able to in the following case:

```python
class AnimalBase(ABC):

    @abstractmethod
    def say(self) -> str:
        ...


class Dog(AnimalBase):

    def say(self, loud=False) -> str:
        return "WOOF" if loud else "woof"
 
class Starfish(AnimalBase):

    def say(self, full_armed=True) -> str:
        return ""
        
 
 def animals_say_loudly(animals: List[AnimalBase]):
     for animal in animals:
         animal.say(loud=True) # Will fail on Starfish
```

To be noted we already created ``argument-renamed`` (#4347) which also weaken LSP (LSP property aren't true when using kwargs anymore) so following that precedent, we might raise a new message for non LSP followers like ``optional-argument-added``.
